x-pack/plugins/actions/server/sub_action_framework/types.ts

```diff
@@ -34,6 +34,25 @@ export type IServiceAbstract<Config, Secrets> = abstract new (
   params: ServiceParams<Config, Secrets>
 ) => SubActionConnector<Config, Secrets>;
 
+export enum ValidatorType {
+  CONFIG,
+  SECRETS,
+}
+
+interface Validate<T> {
+  validate: ValidateFn<T>;
+}
+
+export type ValidateFn<T> = (obj: T, services: ValidatorServices) => void;
```

Comment:
Maybe we can use `ValidatorType['customValidator']<T>` that is defined `x-pack/plugins/actions/server/types.ts`

Reply:
Ah great point.